### PR TITLE
feat: Skip reacting to immutable fields.

### PR DESCRIPTION
**Pull request by stephenh**

### CODE REVIEW

File: packages/integration-tests/src/entities/Author.ts [hunk 1]
```diff
@@ -122,14 +123,19 @@ config.addRule("books", (a) => {
   }
 });
 
-// Example of rule that won't get run unless field is set
+// Example of rule that is always run even if the field is not set
```
Comment (stephenh) [inline packages/integration-tests/src/entities/Author.ts:126]: This behavior changed in the main field-level validation PR, but comment wasn't updated

File: packages/integration-tests/src/reactiveHints.test.ts [hunk 2]
```diff
@@ -93,6 +93,26 @@ describe("reactiveHints", () => {
     ]);
   });
 
+  it("can do read-only string hint", () => {
```
Comment (stephenh) [inline packages/integration-tests/src/reactiveHints.test.ts:96]: These are some extra tests I happened to have locally, unrelated to the current change

File: packages/integration-tests/src/entities/Author.ts [hunk 3]
```diff
@@ -122,14 +123,19 @@ config.addRule("books", (a) => {
...
 config.addRule("mentor", (a) => {
   a.entity.mentorRuleInvoked++;
 });
 
 // Example of cannotBeUpdated
 config.addRule(cannotBeUpdated("age"));
 
+// Example of a rule against an immutaable field
```
Comment (chr1sjf0x) [inline packages/integration-tests/src/entities/Author.ts:134]: quibble: 
```suggestion
// Example of a rule against an immutable field
```

**Review by chr1sjf0x — APPROVED** [review]
Looks good 👍  🍣

**Pull request merged**